lib/onelogin/ruby-saml/utils.rb

```diff
@@ -1,44 +1,83 @@
 module OneLogin
   module RubySaml
+
+    # SAML2 Auxiliary class
+    #
     class Utils
+      # Return the x509 certificate string formatted
+      # @param cert [String] The original certificate 
+      # @param heads [Boolean] If true, the formatted certificate will include the
+      #                        "BEGIN CERTIFICATE" header and the footer.
+      # @return [String] The formatted certificate
+      #
       def self.format_cert(cert, heads=true)
-        cert = cert.delete("\n").delete("\r").delete("\x0D")
-        if cert
-          cert = cert.gsub('-----BEGIN CERTIFICATE-----', '')
-          cert = cert.gsub('-----END CERTIFICATE-----', '')
-          cert = cert.gsub(' ', '')
-
+        unless cert.empty?
```

Comment:
sure! that sounds good +1 `empty?` also works but you just need to check for `cert` first:
`if cert && !cert.empty?`